runtime/src/bank/builtins/core_bpf_migration/target_builtin.rs

```diff
@@ -0,0 +1,252 @@
+use {
+    super::{error::CoreBpfMigrationError, CoreBpfMigrationTarget},
+    crate::bank::Bank,
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        bpf_loader_upgradeable::get_program_data_address,
+        native_loader::ID as NATIVE_LOADER_ID,
+        pubkey::Pubkey,
+    },
+};
+
+/// Used to validate a built-in program's account before migrating to Core BPF.
+#[derive(Debug)]
+pub(crate) struct TargetProgramBuiltin {
+    pub program_address: Pubkey,
+    pub program_account: AccountSharedData,
+    pub program_data_address: Pubkey,
+    pub total_data_size: usize,
+}
+
+impl TargetProgramBuiltin {
+    /// Create a new migration configuration for a built-in program.
+    pub(crate) fn new_checked(
+        bank: &Bank,
+        program_id: &Pubkey,
+        migration_target: &CoreBpfMigrationTarget,
+    ) -> Result<Self, CoreBpfMigrationError> {
+        let program_address = *program_id;
```

Comment:
Recommend removing this line (dereference) to avoid a potentially unnecessary copy, since the only non-error users of this variable before constructing the new struct take a reference.
If you want this to be called `program_address`, update it in the method signature.

Reply:
Sounds good I've updated this function.